### PR TITLE
Import a modified version for Alire.Config 

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**

### CODE REVIEW

File: src/clic-subcommand-instance.adb [hunk 1]
```diff
@@ -635,7 +635,7 @@ package body CLIC.Subcommand.Instance is
             Error_Exit (1);
          end if;
 
-         if not Cmd.Switches_As_Args then
```
Comment (Fabien-Chouteau) [inline src/clic-subcommand-instance.adb:638]: @mosteo you were right.

File: src/clic-config.ads [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+   type Check_Import is
+     access function (Key : Config_Key; Value : TOML.TOML_Value)
+                      return Boolean;
+   --  Return False when a Key/Value combination is not valid. Can be used to
+   --  check formating of string value like email address for instance.
+
+   procedure Import (This   : in out Instance;
+                     Table  :        TOML.TOML_Value;
+                     Origin :        String;
```
Comment (mosteo) [inline src/clic-config.ads:36]: I'd document what this Origin is/does

File: src/clic-config-edit.adb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+   begin
+
+      --  Create the directory for the config file, in case it doesn't exists
+      Create_Path (Containing_Directory (Path));
+
+      Create (File, Out_File, Path);
+      Trace.Debug ("Write config: '" & TOML.Dump_As_String (Table) & "'");
+      Put (File, TOML.Dump_As_String (Table));
+      Close (File);
+   end Write_Config_File;
```
Comment (mosteo) [inline src/clic-config-edit.adb:49]: I think running this when out of space will result on losing your whole config, so I'd use `AAA.Directories.Replacer` here.

**Review by mosteo — APPROVED** [review]
So the idea is to replace Alire.Config with this one, right?

**Comment by Fabien-Chouteau** [comment]
> So the idea is to replace Alire.Config with this one, right?

Yes, I have the Alire changes in a branch already: https://github.com/alire-project/alire/tree/refactoring/clic_config.
But it might be better to wait for after the 1.1.0 release.

**Review by mosteo — APPROVED** [review]

**Comment by mosteo** [comment]
> Yes, I have the Alire changes in a branch already: https://github.com/alire-project/alire/tree/refactoring/clic_config.
> But it might be better to wait for after the 1.1.0 release.

I'll create a `next` branch into which we merge post-1.1 ready PRs, so I don't mix things up, and we can keep using `master` for things that will end in 1.1

**Pull request merged**